hibernate-core/src/main/java/org/hibernate/cfg/BytecodeSettings.java

```diff
@@ -28,6 +28,17 @@ public interface BytecodeSettings {
 	@Deprecated( forRemoval = true )
 	String BYTECODE_PROVIDER = "hibernate.bytecode.provider";
 
+	/**
+	 * This is similar to the now deprecated legacy property {@code hibernate.bytecode.provider} except
+	 * it's used specifically to pass an existing instance of a {@link org.hibernate.bytecode.spi.BytecodeProvider};
+	 * this happens to also allow to override the implementation, but is primarily intended to allow reusing a
+	 * specific instance; this could be useful when the implementation benefits from internal caches.
+	 * When not set, Hibernate will create its default implementation.
+	 *
+	 * @settingDefault {@code null}
+	 */
+	String BYTECODE_PROVIDER_INSTANCE = "hibernate.enhancer.bytecodeprovider.instance";
```

Comment:
Are you slipping in performance improvements while you're at it? :)

Reply:
ahah :)  Like I was saying in zulip, I initially thought this was necessary. It likely isn't, but I haven't had the chance to test it w/o it.
It definitely should get its own JIRA but I'd still suggest we include it, what's currently happening is madness: BB gets initialized once for each class X each thread X each persistence unit. And its quite heavy.